webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -299,6 +376,152 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
+      return;
+    } 
+    return xmlDoc; 
+  }
+
+  private getFormValues(xmlDoc: Document, propPaths: string[]){
+    const values:string[] = [];
+    for (const path of propPaths){
+      try {
+        const value = xmlDoc.evaluate(
+          path,
+          xmlDoc,
+          null,
+          XPathResult.STRING_TYPE,
+          null
+        ).stringValue;
+        values.push(value);
+      } catch (error) {
+        console.error(`Path ${path} value could not be resolved!`);
+      }
+    }
+    return values;
+  }
+
+  private queryPropInfo(xmlDoc: Document, info: Exclude<QueryParamsInfoType, undefined>){
+    const propPaths = info.valuePaths;
+    const values = this.getFormValues(xmlDoc, propPaths);
+    const result = info.query(...values);
+    return result;
+  }
+
+  private calculateSiblingLikenessScore(sibling: Sibling, props: { value: string; compareToDocProp: string }[]) {
+    let totalScore = 0;
+    let count = 0;
+
+    for (const field of props) {
+      if (Object.prototype.hasOwnProperty.call(sibling, field.compareToDocProp)) {
+        const test = this.strategy(field.value.toLowerCase(), sibling[field.compareToDocProp].toLowerCase());
+        totalScore += test;
+        count++;
+      }
+    }
+
+    return count > 0 ? totalScore / count : null;
+  }
+
+  private getLikelyDuplicates(siblings: Sibling[], props: {value: string; compareToDocProp: string}[]){
+    const duplicates: Array<Duplicate> = [];
+    for (const sibling of siblings){
+      const score = this.calculateSiblingLikenessScore(sibling, props);
+      if (score !== null && score < this.duplicateThreshold){
+        duplicates.push({_id: sibling._id, name: sibling.name, reported_date: sibling.reported_date, score});
+      }
+    }
+
+    return duplicates;
+  }
+
+  // Normalize the distance by dividing by the length of the longer string. 
+  // This can make the metric more adaptable across different string lengths
+  private normalizedLevenshtein(str1: string, str2: string) {
+    const distance = this.levenshteinDistance(str1, str2);
+    const maxLen = Math.max(str1.length, str2.length);
+    return (maxLen === 0) ? 0 : (distance / maxLen);
+  }
+
+  // The Levenshtein distance is a measure of the number of edits (insertions, deletions, and substitutions) 
+  // required to change one string into another.
+  private levenshteinDistance(str1: string, str2: string) {
+    return new Levenshtein(str1, str2).distance;
+  }
+
+  // Note: Look into 'https://www.npmjs.com/package/fuzzball' to perform various matching calculations
+
+  private cleanUpDuplicateSection() {
+    const $duplicateInfoElement = $('#contact-form').find('#duplicate_info');
+    $duplicateInfoElement.empty(); // Remove all child nodes
+    $duplicateInfoElement.hide();
+  }
+
+  // Promise.allSettled is not available due to the app's javascript version
+  private allSettledFallback(promises: Promise<Exclude<any, null | undefined>>[]): Promise<ReturnType[]> {
```

Comment:
@jkuester @ChinHairSaintClair  Would promise `allSettled` not be transpiled to a target version?